### ISSUE TITLE
Data quality issues

### ISSUE THREAD

**Issue by zifangu** [NSAPH-Data-Platform/nsaph-cms#4]
When querying for Medicaid data, we found two main issues: invalid zipcode and multiple admissions.
## zipcode
Out of the unique zipcodes pulled, not all can be mapped to a valid location in the US. Would this come from CMS itself or during the curation process? (happy to help with the latter)
This is something minor, but the zipcode column sometimes only have 3 or 4 digits. Are we to assume those zipcodes should have leading 0s before them?

## multiple admissions
This came from doing some EDA. We discovered that within the same person (identified by bene_id), he/she could have different diagnoses and discharge dates. Any suggestions on how this could happen?

Looking forward!

**Comment by daniellebraun** [comment]
@mmcentre

**Comment by mmcentre** [comment]
@daniellebraun @zifangu 
For Medicaid data, zip code should be taken from Enrollments table. See either 
[platform documentation](https://nsaph-data-platform.github.io/nsaph-platform-docs/pipelines/cms/doc/Medicaid.html#enrollments)
or [source code documentation](https://github.com/NSAPH-Data-Platform/nsaph-cms/blob/develop/doc/Medicaid.md#enrollments) for details. Enrollments table has a column named `zip` of type integer. (Using integer type helps with query performance and also eliminates the problem of whitespaces in the text).

I have run the following query to see how many invalid zip codes we have:

```
SELECT 
    CASE WHEN 
        EXISTS (SELECT zip FROM public.hud_zip2fips AS h WHERE h.zip = e.zip)
        THEN 1
        ELSE 0
    END AS valid_zip,
    COUNT(*)     
FROM 
    medicaid.enrollments AS e
GROUP BY 1
ORDER BY 1
```

The result is:

- 7,866,660 records with invalid zip codes
- 775,075,529 records with valid zip codes

In other words, we have about 1% of records with invalid zip code. This is consistent with the overall quality of Medicaid data.

Out of about 8 million of invalid zips, 5,312,405 records have zip 0, i.e. missing. Hence the percentage of records with really invalid code is about 0.3%. I think this is not bad.

Another query that you might find useful:

```
SELECT 
    zip,
    COUNT(*),
    CASE WHEN 
        EXISTS (SELECT zip FROM public.hud_zip2fips AS h WHERE h.zip = e.zip)
        THEN 1
        ELSE 0
    END AS valid_zip     
FROM 
    medicaid.enrollments AS e
GROUP BY zip
ORDER BY 3, 1

```

**Comment by daniellebraun** [comment]
thanks, super helpful, @zifangu does this make sense?

**Comment by zifangu** [comment]
@daniellebraun Yes, it does. 

@mmcentre Thank you for providing this logic. Invalid admission dates seem to be a common issue in these files. 

Following this guidance, the following query showed us how many multiple admission dates we have:

```
SELECT  
  CASE  WHEN ct > 1
  THEN 1
  ELSE 0
  END AS multiple_amdission,
  COUNT(*)
FROM  
(
SELECT  count(bene_id) AS  ct, bene_id, admission_date FROM  medicaid.admissions
GROUP  BY  admission_date, bene_id
) AS  multi_admin
GROUP BY 1
ORDER BY 1;
```

- 99,716,822 records of bene_id has unique admission dates
- 1,154,620 records of bene_id has duplicate admission dates

This translates to 1.2% of total admissions records having duplicate admission dates. It appears to me that this is consistent with the overall quality of Medicaid data, if we follow the same threshold for zip code invalidness). But happy to discuss further.

**Comment by mmcentre** [comment]
I believe this issue has been resolved. @zifangu please feel free to open a new one for any remaining questions
